src/Engine/Renderer/Texture/Texture.hpp

```diff
@@ -1,237 +1,251 @@
+ */
+class RA_ENGINE_API Texture final {
+  public:
+    /** Texture parameters
+     */
+    TextureParameters m_textureParameters;
+
+    /** Textures are not copyable, delete copy constructor.
+     */
+    Texture( const Texture& ) = delete;
+
+    /** Textures are not copyable, delete operator =.
+     */
+    void operator=( const Texture& ) = delete;
+
+    /**
+     * Texture constructor. No OpenGL initialization is done there.
+     *
+     * @param texParameters Name of the texture
+     */
+    explicit Texture( const TextureParameters& texParameters );
+
+    /**
+     * Texture destructor. Both internal data and OpenGL stuff are deleted.
+     */
+    ~Texture();
+
+    /** @brief Generate the OpenGL representation of the texture according to the stored TextureData
+     *
+     * This method use the available m_textureParameters to generate and configure OpenGL
+     * texture.
+     *
+     * Before uploading texels to the GPU, this method will apply RGB space conversion if needed.
+     *
+     * @param linearize (default false) : convert the texture from sRGB to Linear RGB color space
+     * before OpenGL initialisation
+     * @note This will become soon the only way to generate an Radium Engine OpenGL texture.
+     */
+    void initializeGL( bool linearize = false );
+
+    /**
+     * @brief Init the textures needed for the cubemap from OpenGL point of view.
+     *
+     * Generate, bind and configure OpenGL texture.<br/>
+     * Also sets wrapping to GL_REPEAT and min / mag filters to GL_LINEAR, although no mipmaps are
+     * generated.<br/><br/>
+     *
+     * It is highly recommended to take a look at
+     * <a href="https://www.opengl.org/wiki/GLAPI/glTexImage2D">glTexImage2D documentation</a>
+     * since this method doc will highly refer to it.
+     *
+     *
+     * @param width Width of the six 2D textures.
+     *
+     * @param height Height of the six 2D textures.
+     *
+     * @param format The format of the pixel data.
+     * Refer to the link given above, at the \b format section
+     * for further informations about the available formats.
+     *
+     * @param data Data contained in the texture. Can be nullptr. <br/>
+     * If \b data is not null, the texture will take the ownership of it.
+     *
+     * @param linearize (default false) : convert the texture from sRGB to Linear RGB color space
+     *
+     * @param mipmaped (default false) : generate a prefiltered mipmap for the texture.
+     *
+     * @todo integrate this method in the same workflow than other textures ...
+     */
+    void generateCube( uint width, uint height, GLenum format, void** data = nullptr,
+                       bool linearize = false, bool mipmaped = false );
+
+    /**
+     * @brief Bind the texture to enable its use in a shader
+     * @param unit Index of the texture to be bound. If -1 only calls glBindTexture.
+     */
+    void bind( int unit = -1 );
+
+    /**
```

Comment:
```suggestion
    /**
     * Bind the texture to an image unit for the purpose of reading and writing it from shaders.
     * @note, only available since openGL 4.2, not available on MacOs
```